src/pages/Profile.js

```diff
@@ -245,6 +246,78 @@ const FenceLink = ({ provider: { key, name, expiresAfter, short } }) => {
 }
 
 
+const PassportLinker = ({ queryParams, provider, prettyName }) => {
+  const signal = useCancellation()
+  const [accountInfo, setAccountInfo] = useState()
+  const [authUrl, setAuthUrl] = useState()
+
+  useOnMount(() => {
+    const loadAuthUrl = withErrorReporting(`Error loading ${prettyName} account link URL`, async () => {
+      setAuthUrl(await Ajax(signal).User.externalAccount(provider).getAuthUrl())
+    })
+    const loadAccount = withErrorReporting(`Error loading ${prettyName} account`, async () => {
+      setAccountInfo(await Ajax(signal).User.externalAccount(provider).get())
+    })
+    const linkAccount = withErrorReporting(`Error linking ${prettyName} account`, async code => {
+      setAccountInfo(await Ajax().User.externalAccount(provider).linkAccount(code))
+    })
+
+    loadAuthUrl()
+
+    if (Nav.getCurrentRoute().name === 'ecm-callback') {
+      const { code, state } = queryParams
+      const { provider: linkProvider } = JSON.parse(atob(state))
+
+      if (provider === linkProvider) {
+        window.history.replaceState({}, '', `/${Nav.getLink('profile')}`)
+
+        linkAccount(code)
+        return
+      }
+    }
+
+    loadAccount()
+  })
+
+  const unlinkAccount = withErrorReporting(`Error unlinking ${prettyName} account`, async () => {
+    setAccountInfo(undefined)
+    await Ajax().User.externalAccount(provider).unlink()
+    setAccountInfo(null)
+  })
+
+  return div({ style: styles.idLink.container }, [
+    div({ style: styles.idLink.linkContentTop(false) }, [
+      h3({ style: { marginTop: 0, ...styles.idLink.linkName } }, [prettyName]),
+      Utils.cond(
```

Comment:
Point to consider for the future-- making the HTML more semantic so it is more clearly a list of external accounts.